src/ListView.vala

```diff
@@ -84,55 +121,54 @@ public class Tasks.ListView : Gtk.Grid {
         });
 
         notify["source"].connect (() => {
+            remove_views ();
+
             foreach (unowned Gtk.Widget child in task_list.get_children ()) {
                 child.destroy ();
             }
 
-            if (source != null) {
-                update_request ();
-
-                try {
-                     var client = (ECal.Client) ECal.Client.connect_sync (source, ECal.ClientSourceType.TASKS, -1, null);
-
-                     var task_row = new Tasks.TaskRow.for_source (source);
-                     task_row.task_save.connect ((task) => {
-                         add_task (client, task);
-                     });
-                     task_list.add (task_row);
-
-                     /*
-                      * We need to pass a valid S-expression to guarantee the below callback events are fired.
-                      *
-                      * See `e-cal-backend-sexp.c` of evolution-data-server for available S-expressions:
-                      * https://gitlab.gnome.org/GNOME/evolution-data-server/-/blob/master/src/calendar/libedata-cal/e-cal-backend-sexp.c
-                      */
-                     client.get_view_sync ("(contains? 'any' '')", out view, null);
-
-                     view.objects_added.connect ((objects) => on_objects_added (source, client, objects));
-                     view.objects_removed.connect ((objects) => on_objects_removed (source, client, objects));
-                     view.objects_modified.connect ((objects) => on_objects_modified (source, client, objects));
-
-                     view.start ();
-
-                 } catch (Error e) {
-                     critical (e.message);
-                 }
-            } else {
-                editable_title.text = "";
-            }
-
+            update_request ();
             show_all ();
         });
 
         editable_title.changed.connect (() => {
+            if (source == null) {
+                return;
+            }
             source.display_name = editable_title.text;
             source.write.begin (null);
         });
+
+        update_request ();
     }
 
     public void update_request () {
-        editable_title.text = source.dup_display_name ();
-        Tasks.Application.set_task_color (source, editable_title);
+        if (source == null) {
+            editable_title.sensitive = false;
+            editable_title.text = _("Scheduled");
+            settings_button_revealer.reveal_child = false;
+
+            Tasks.Application.set_task_color_from_string ("@colorAccent", editable_title);
+
+            task_list.@foreach ((row) => {
+                if (row is Tasks.TaskRow) {
+                    (row as Tasks.TaskRow).update_request ();
```

Comment:
The compiler doesn't like this: `Access to possible 'null'. Perform a check or use an unsafe cast.`

Since we do check if the row is a type of `Tasks.TaskRow` we can probably cast like:
```suggestion
                    ((Tasks.TaskRow) row).update_request ();
```